src/routes/assemblies/[id]/members/[groupby]/+page.svelte

```diff
@@ -1,8 +1,243 @@
+		isMobile = !showFilter;
+		if (showFilter) {
+			currentCatg = (
+				groups[0].name +
+				'-' +
+				('subgroups' in groups[0] ? groups[0].subgroups[0].name : '')
+			).replace(/ /g, '-');
+
+			const scroller = scrollama();
+
+			scroller
+				.setup({
+					step: '.member-subcategory',
+					// @ts-expect-error Documentation บอกว่าใช้ string ที่มี px ได้ https://github.com/russellsamora/scrollama#:~:text=number%20(0%20%2D%201%2C%20or%20string%20with%20%22px%22)
+					offset: '128px'
+				})
+				.onStepEnter((response) => {
+					currentCatg = response.element.children[0].id;
+				});
+
+			return scroller.destroy;
+		}
+	});
 </script>
 
-<div class="whitespace-pre">
-	{JSON.stringify(groups, undefined, 2)}
+<Header {data} bind:searchQuery {mounted} />
+<Tab {data} />
+<div class="relative flex">
+	{#if showFilter}
+		<aside
+			class={[
+				'flex flex-col gap-4 px-6 bg-white',
+				'fixed left-0 right-0 top-0 bottom-0 z-40',
+				'md:sticky md:top-[80px] md:w-[250px] md:h-[calc(100dvh-80px)] md:flex-none md:z-0',
+				'member-aside'
+			].join(' ')}
+		>
+			<div class="flex -mr-6 md:mr-0">
+				<Search
+					class="flex-1 {!mounted ? '-mt-6' : ''}"
+					placeholder="ค้นหาชื่อบุคคล"
+					light
+					bind:value={searchQuery}
+					skeleton={!mounted}
+				/>
+				<div class="md:hidden flex">
+					<Button
+						kind="ghost"
+						icon={Minimize}
+						iconDescription="ซ่อนตัวเลือก"
+						tooltipPosition="right"
+						tooltipAlignment="end"
+						on:click={() => {
+							showFilter = false;
+						}}
+						skeleton={!mounted}
+					/>
+				</div>
+			</div>
+			{#if currentPath === 'party'}
+				<FormGroup legendText="ประเภท" noMargin>
+					<div class="flex items-center justify-between overflow-hidden">
+						<Checkbox
+							labelText="แบ่งเขต"
+							class="!m-0"
+							bind:checked={isByDistrict}
+							skeleton={!mounted}
+						/>
+						<Checkbox
+							labelText="บัญชีรายชื่อ"
+							class="!m-0"
+							bind:checked={isByPartylist}
+							skeleton={!mounted}
+						/>
+					</div>
+				</FormGroup>
+			{/if}
+			<div class="flex-[1_1_auto] h-0 overflow-y-auto">
+				<Accordion class="accordion-content-full" skeleton={!mounted}>
+					{#each filteredGroup as group (group.name)}
+						<AccordionItem open={currentPath === 'party'}>
+							<span slot="title" class="font-semibold"
+								>{group.name}
+								{#if !['party', 'province'].includes(currentPath) && 'subgroups' in group}
+									<span class="font-normal text-gray-60"
+										>({group.subgroups.map((e) => e.members.length).reduce((a, c) => a + c)})</span
+									>
+								{/if}
+							</span>
+							{#if 'subgroups' in group}
+								<ul class="flex flex-col">
+									{#each group.subgroups as { name, members, icon } (name)}
+										<li class="border-b border-b-solid border-b-gray-30 last:border-none">
+											<a
+												href="#{group.name.replace(/ /g, '-')}-{name.replace(/ /g, '-')}"
```

Comment:
replaceAll is easier to understand and safely to use now https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll

But can you concat before replace? (Or there is some case I didn't think about.)

```js
`${group.name}-${name}`.replaceAll(' ', '-')
```

Reply:
And I just found several usages below, better refactor to a function like

```js
const getSubgroupHeadingId = (group, name) => `${group.name}-${name}`.replaceAll(' ', '-')
```